02-peer-protocol.md

```diff
@@ -662,119 +665,119 @@ of the receiving node to change the `scriptpubkey`.
-
-The receiving node:
-  - if the `signature` is not valid for either variant of closing transaction
-  specified in [BOLT #3](03-transactions.md#closing-transaction) OR non-compliant with LOW-S-standard rule<sup>[LOWS](https://github.com/bitcoin/bitcoin/pull/6769)</sup>:
-    - MUST send a `warning` and close the connection, or send an
-      `error` and fail the channel.
-  - if `fee_satoshis` is equal to its previously sent `fee_satoshis`:
-    - SHOULD sign and broadcast the final closing transaction.
-    - MAY close the connection.
-  - if `fee_satoshis` matches its previously sent `fee_range`:
-    - SHOULD use `fee_satoshis` to sign and broadcast the final closing transaction
-    - SHOULD reply with a `closing_signed` with the same `fee_satoshis` value if it is different from its previously sent `fee_satoshis`
-    - MAY close the connection.
-  - if the message contains a `fee_range`:
-    - if there is no overlap between that and its own `fee_range`:
-      - SHOULD send a warning
-      - MUST fail the channel if it doesn't receive a satisfying `fee_range` after a reasonable amount of time
+Note: the details and requirements for the transaction being signed are in [BOLT 3](03-transactions.md#closing-transaction)).
+
+An output is *dust* if:
+- It is P2SH and the amount is < 540 satoshis
+- It is P2WSH or P2TR and the amount is < 330 satoshis
+- (No OP_RETURN of any amount is dust)
+
+Both nodes:
+  - After a `shutdown` has been received, AND no HTLCs remain in either commitment transaction:
+    - SHOULD send a `closing_complete` message.
+
+The sender of `closing_complete` (aka. "the closer"):
+  - MUST set `fee_satoshis` to a fee less than or equal to its outstanding balance, rounded down to whole satoshis.
+  - MUST set `fee_satoshis` so that at least one output is not dust.
+  - MUST use the last send and received `shutdown` `scriptpubkey` to generate the closing transaction specified in [BOLT #3](03-transactions.md#closing-transaction).
+  - If it sets `signature` fields, MUST set them as valid signature using its `funding_pubkey` of:
+    - `closer_noclosee`: closing transaction with only the local ("closer") output.
+    - `no_closer_closee`: closing transaction with only the remote ("closee") output.
+    - `closer_and_closee`: closing transaction with both the closer and closee outputs.
+  - If the local outstanding balance (in millisatoshi) is less than the remote outstanding balance:
+    - MUST NOT set `closer_no_closee`.
+    - MUST set exactly one of `no_closer_closee` or `closer_and_closee`.
+    - MUST set `no_closer_closee` if the local output amount is dust.
+    - MAY set `no_closer_closee` if it considers the closee output amount uneconomic AND its `scriptpubkey` is not `OP_RETURN`.
+  - Otherwise (not lesser amount, cannot remove own output):
+    - MUST NOT set `no_closer_closee`.
+    - MUST set both `closer_no_closee` and `closer_and_closee`.
```

Comment:
Why require setting both here? If the closee's output is below dust, we should be allowed to only set `closer_no_closee`, because `closer_and_closee` would lead to a non-standard transaction?

I understand that when the closee's output is *not* below dust, we must still set `close_no_closee` to allow them to omit their output though.

So I think the requirements for that section should be:

```
 - Otherwise (not lesser amount, cannot remove own output):
    - MUST NOT set `no_closer_closee`.
    - If the closee's output amount is dust:
      - MUST set `closer_no_closee`.
      - SHOULD NOT set `closer_and_closee`.
    - Otherwise:
      - MUST set both `closer_no_closee` and `closer_and_closee`.
```